README.md

```diff
@@ -1,38 +1,315 @@
+[![Build Status](https://travis-ci.org/esl/sparrow.svg?branch=master)](https://travis-ci.org/esl/sparrow)
+[![Coverage Status](https://coveralls.io/repos/github/esl/sparrow/badge.svg)](https://coveralls.io/github/esl/sparrow)
 
-**TODO: Add description**
+Currently it provides support for the following APIs:
+* [FCM v1](https://firebase.google.com/docs/cloud-messaging/)
+* [APNS](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/APNSOverview.html#//apple_ref/doc/uid/TP40008194-CH8-SW1)
 
-**TODO: Redirect Erlang lager to Elixir logger**
+**Requirements**
+* Elixir 1.6 or higher
+* Erlang OTP 20.3.6 or OTP 21
 
-**TODO: Add publishing package to hex** 
+# Build sparrow config
+This section describes how to write a config file for Sparrow.
+If you wish to use just one of the following services, do not include the other in the config.
 
-**TODO: Check API diffs between these articles:** 
+```elixir
+config :sparrow, config: [fcm_config, apns_config]
+```
+**Config opts**
 
-[link](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/generating_a_remote_notification)
-and
-[link](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html#//apple_ref/doc/uid/TP40008194-CH17-SW1)
+***FCM specific***
+- `:path_to_json` - is a path to a json file provided by FCM
 
-## Installation
+***APNS specific***
+- `:auth_type` - defines the authentication type, the allowed values are: `:token_based`, `:certificate_based`
 
-[![Build Status](https://travis-ci.com/esl/sparrow.svg?branch=master)](https://travis-ci.com/esl/sparrow)
-[![Coverage Status](https://coveralls.io/repos/github/esl/sparrow/badge.svg)](https://coveralls.io/github/esl/sparrow)
+    1 `:token_based` requres to set:
+    - `:token_id` - is a unique atom referring to a token with the same `:token_id` in config
+    
+    2 `:certificate_based` requres to set:
+    
+    - `:cert` - path to the certificate file provided by APNS
+    - `:key` - path to the key file provided by APNS
+
+***H2Worker config***
+- `:endpoint` - service uri address 
+- `:port` - service port
+- `:tls_opts` - passed to erlang [ssl](http://erlang.org/doc/man/ssl.html) module (see DATA TYPES -> ssl_option())
+- `:ping_interval` - number of miliseconds between each [ping](https://http2.github.io/http2-spec/#PING)
+- `:reconnect_attempts` - number of attempts to connect before failing the connection
+
+***Pool config***
+- `:tags` - see the [tags](#Tags) section
+- `:worker_num` - number of workers in a pool
+- `:raw_opts` - opts passed directly to [wpool](github.com/inaka/worker_pool)
+
+**FCM config**
+```elixir
+fcm_config = [fcm:
+    [
+        [
+            # Authentication
+            path_to_json: "path/to/fcm_token.json", # mandatory, path to FCM authentication JSON file
```

Comment:
I see that at this point you fell back to describing config opts inline of an example. This is (in my opinion) much more readable than the previous sections - so what's the use of the previous sections? Why are they there?

Reply:
I wanted to describe config options like in:
https://github.com/devinus/poolboy#options
or
https://github.com/inaka/worker_pool#starting-a-pool

Reply:
And show "super example" in the same time. Hoping that both combined will be more readable.